x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClientManager.java

```diff
@@ -89,6 +91,12 @@ public class HttpClientManager implements Closeable {
         Setting.Property.Dynamic
     );
 
+    public static final Setting<TimeValue> CONNECTION_TIMEOUT = Setting.timeSetting(
```

Comment:
How about we move this into `HttpSettings`. I don't think we need to add a `addSettingsUpdateConsumer` because we'd have to close the client and recreate it but I think that class was for the settings for the client.